sql: remove global zone configs on final DROP REGION

otan opened the pull request:
Release note (bug fix): Fix a bug where using DROP REGION on the last
region of the given multi-region database would not delete the global
zone configurations for LOCALITY GLOBAL tables.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/62121)
<!-- Reviewable:end -->

ajstorm reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/62121)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan)

___
*[pkg/ccl/logictestccl/testdata/logic_test/multi_region, line 1087 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/62121#-MW-g2nI1yaC0lG7Zfqn:-MW-g2nI1yaC0lG7Zfqo:bqzgb4p) ([raw file](https://github.com/cockroachdb/cockroach/blob/22ff5ade6577816f41a373dab8fb2424e6ea14e9/pkg/ccl/logictestccl/testdata/logic_test/multi_region#L1087)):*
> ```
>                          num_replicas = 3,
>                          num_voters = 3,
>                          constraints = '{+region=ca-central-1: 1}',
> ```

wha?  Shouldn't this be gone?


<!-- Sent from Reviewable.io -->

otan reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/62121)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ajstorm)

___
*[pkg/ccl/logictestccl/testdata/logic_test/multi_region, line 1087 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/62121#-MW-g2nI1yaC0lG7Zfqn:-MW0TNPqA4NEl4VsjfTX:b52fsdp) ([raw file](https://github.com/cockroachdb/cockroach/blob/22ff5ade6577816f41a373dab8fb2424e6ea14e9/pkg/ccl/logictestccl/testdata/logic_test/multi_region#L1087)):*
<details><summary><i>Previously, ajstorm (Adam Storm) wrote…</i></summary><blockquote>

wha?  Shouldn't this be gone?
</blockquote></details>

I need to rebase... 


<!-- Sent from Reviewable.io -->

ajstorm reviewed (approved):
> Reviewed 3 of 3 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/62121)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan)

___
*[pkg/ccl/logictestccl/testdata/logic_test/multi_region, line 1087 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/62121#-MW-g2nI1yaC0lG7Zfqn:-MW4FEBY2d96UlwCJN58:b-pkz8ap) ([raw file](https://github.com/cockroachdb/cockroach/blob/22ff5ade6577816f41a373dab8fb2424e6ea14e9/pkg/ccl/logictestccl/testdata/logic_test/multi_region#L1087)):*
<details><summary><i>Previously, otan (Oliver Tan) wrote…</i></summary><blockquote>

I need to rebase... 
</blockquote></details>

muy bueno
___
*[pkg/sql/alter_database.go, line 394 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/62121#-MW4FgZLEH3Qm0zYuadv:-MW4FgZLEH3Qm0zYuadw:b3610nz) ([raw file](https://github.com/cockroachdb/cockroach/blob/7111b9d98d2c74a4e0eedc2033e5d383e96fbeac/pkg/sql/alter_database.go#L394)):*
> ```Go
> 				if t.RegionalByTable.Region != nil {
> 					return errors.AssertionFailedf(
> 						"unexpected REGIONAL BY ROW IN <region> on table %s during DROP REGION",
> ```

`REGIONAL BY TABLE`?

Oh, and _there_ you have it.  The introduction of a table name in an AssertionFailedF :-P.  I think we need to use `errors.Safe` here.
___
*[pkg/sql/alter_database.go, line 401 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/62121#-MW4H-Qc16B486Rtk6U4:-MW4H-Qd-115qzzf_H-k:b-98iwke) ([raw file](https://github.com/cockroachdb/cockroach/blob/7111b9d98d2c74a4e0eedc2033e5d383e96fbeac/pkg/sql/alter_database.go#L401)):*
> ```Go
> 				// REGIONAL BY ROW drops should have errored earlier.
> 				// Any new localities fall into this default clause.
> 				return errors.AssertionFailedf(
> ```

Can we add an assertion for RBR here too?  Or catch it along with the assertion above?


<!-- Sent from Reviewable.io -->

otan commented:
___
*[pkg/sql/alter_database.go, line 394 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/62121#-MW4FgZLEH3Qm0zYuadv:-MW4JC0Q7DirSwqfpDKs:b-vbs4rf) ([raw file](https://github.com/cockroachdb/cockroach/blob/7111b9d98d2c74a4e0eedc2033e5d383e96fbeac/pkg/sql/alter_database.go#L394)):*
<details><summary><i>Previously, ajstorm (Adam Storm) wrote…</i></summary><blockquote>

`REGIONAL BY TABLE`?

Oh, and _there_ you have it.  The introduction of a table name in an AssertionFailedF :-P.  I think we need to use `errors.Safe` here.
</blockquote></details>

Wrong way around. We only want to use errors.Safe if it is not PII. This is. 


<!-- Sent from Reviewable.io -->

ajstorm commented:
Ahhh, my bad.  All good then, minus the part about you getting sleep.

On Thu, Mar 18, 2021 at 8:31 AM Oliver Tan ***@***.***> wrote:

> ------------------------------
>
> *pkg/sql/alter_database.go, line 394 at r2
> <https://reviewable.io/reviews/cockroachdb/cockroach/62121#-MW4FgZLEH3Qm0zYuadv:-MW4JC0Q7DirSwqfpDKs:b-vbs4rf>
> (raw file
> <https://github.com/cockroachdb/cockroach/blob/7111b9d98d2c74a4e0eedc2033e5d383e96fbeac/pkg/sql/alter_database.go#L394>):*
> *Previously, ajstorm (Adam Storm) wrote…*
>
> REGIONAL BY TABLE?
>
> Oh, and *there* you have it. The introduction of a table name in an
> AssertionFailedF :-P. I think we need to use errors.Safe here.
>
> Wrong way around. We only want to use errors.Safe if it is not PII. This
> is.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/62121#issuecomment-801887644>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEMXVOS2X3T5MEDLGUTEN5DTEHXCHANCNFSM4ZJWCK5Q>
> .
>

otan committed:
sql: remove global zone configs on final DROP REGION

Release note (bug fix): Fix a bug where using DROP REGION on the last
region of the given multi-region database would not delete the global
zone configurations for LOCALITY GLOBAL tables.

otan commented:
bors r=ajstorm
